parquet/src/test/java/org/apache/iceberg/parquet/TestParquetDataWriter.java

```diff
@@ -266,4 +279,37 @@ public void testInvalidUpperBoundBinary() throws Exception {
     assertThat(dataFile.lowerBounds()).as("Should have a valid lower bound").containsKey(3);
     assertThat(dataFile.upperBounds()).as("Should have a null upper bound").doesNotContainKey(3);
   }
+
+  @Test
+  public void testDataWriterWithVariantShredding() throws IOException {
```

Comment:
Consider adding tests for edge cases:

  1. **Null shredding function**: What happens when `variantShreddingFunc(null)` is called?
  2. **No shredding function set**: Default behavior when the method is never called
  3. **Exception handling**: If the shredding function throws an exception during writing

Reply:
This PR does not cover the implementation of the Variant type writer. The related tests can be found in TestVariantWriters